cirq-core/cirq/devices/grid_device_metadata.py

```diff
@@ -73,17 +73,15 @@ def __init__(
         connectivity.add_edges_from(sorted(pair_set), directed=False)
         super().__init__(flat_pairs, connectivity)
         self._qubit_pairs = frozenset(pair_set)
-        self._supported_gates = supported_gates
+        self._gateset = gateset
 
         if gate_durations is not None:
-            working_gatefamilies = frozenset(
-                g for gset in gate_durations.keys() for g in gset.gates
-            )
-            if working_gatefamilies != supported_gates.gates:
-                missing_items = working_gatefamilies.difference(supported_gates.gates)
+            working_gatefamilies = frozenset(gate_durations.keys())
+            if working_gatefamilies != gateset.gates:
+                missing_items = working_gatefamilies.difference(gateset.gates)
                 raise ValueError(
                     "Supplied gate_durations contains gates not present"
-                    f" in supported_gates. {missing_items} in supported_gates"
+                    f" in gateset. {missing_items} in gateset"
                     " is False."
                 )
```

Comment:
We are checking for equality but reporting difference in the ValueError. It's possible that working_gatefamilies is a subset of gateset.gates, in which case `missing_items` will be an empty set and error message would not be useful.